test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1400,3 +1401,166 @@ var _ = ginkgo.Describe("Pod controller interacting with scheduler", ginkgo.Orde
+
+		cq = testing.MakeClusterQueue("cq").
+			ResourceGroup(*testing.MakeFlavorQuotas(fl.Name).
+				Resource(corev1.ResourceCPU, "9").
+				Resource(corev1.ResourceMemory, "36").
+				Obj()).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, cq)).Should(gomega.Succeed())
+
+		lq = testing.MakeLocalQueue("lq", ns.Name).ClusterQueue(cq.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, lq)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, fl, true)
+	})
+
+	ginkgo.When("pod group not ready", func() {
+		ginkgo.It("should evict workload due workload deactivated on pods ready timeout", func() {
+			podGroupName := "pod-group"
+			pod := testingpod.MakePod("pod", ns.Name).
+				Group(podGroupName).
+				GroupTotalCount("1").
+				Queue(lq.Name).
+				Request(corev1.ResourceCPU, "1").
+				NodeSelector("foo", "doesnotexist").
+				Obj()
+			ginkgo.By("creating pod", func() {
+				gomega.Expect(k8sClient.Create(ctx, pod)).Should(gomega.Succeed())
+			})
+
+			createdWorkload := &kueue.Workload{}
+			wlKey := types.NamespacedName{Name: podGroupName, Namespace: pod.Namespace}
+
+			ginkgo.By("checking that workload was created")
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, createdWorkload)).Should(gomega.Succeed())
+				g.Expect(createdWorkload.Status.Conditions).To(gomega.ContainElements(
+					gomega.BeComparableTo(metav1.Condition{
+						Type:    kueue.WorkloadPodsReady,
+						Status:  metav1.ConditionFalse,
+						Reason:  kueue.WorkloadPodsReady,
+						Message: "Not all pods are ready or succeeded",
+					}, util.IgnoreConditionTimestampsAndObservedGeneration),
+				))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			createdPod := &corev1.Pod{}
+			podKey := client.ObjectKeyFromObject(pod)
+
+			ginkgo.By("setting pod failed")
+			gomega.Expect(k8sClient.Get(ctx, podKey, createdPod)).Should(gomega.Succeed())
+			createdPod.Status.Phase = corev1.PodFailed
+			gomega.Expect(k8sClient.Status().Update(ctx, createdPod)).Should(gomega.Succeed())
+
+			ginkgo.By("deleting pod")
+			gomega.Expect(k8sClient.Delete(ctx, createdPod)).Should(gomega.Succeed())
+
+			ginkgo.By("admit the workload")
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, podKey, createdPod)).Should(gomega.Succeed())
+				g.Expect(createdPod.Status.Phase).Should(gomega.Equal(corev1.PodFailed))
+				g.Expect(createdPod.DeletionTimestamp).ShouldNot(gomega.BeNil())
+				g.Expect(k8sClient.Get(ctx, wlKey, createdWorkload)).Should(gomega.Succeed())
+				g.Expect(createdWorkload.Spec.PodSets).To(gomega.HaveLen(1))
+				createdWorkload.Status.RequeueState = &kueue.RequeueState{
+					Count:     ptr.To[int32](1),
+					RequeueAt: ptr.To(metav1.NewTime(time.Now())),
+				}
```

Comment:
I would prefer not to set these values "by hand" since this is responsibility of the workload_controller, which is part of the test. We expect the count to be bumped to 1 due to eviction due to PodsReady timeout.

Reply:
I think on this case we will have the same flaky proble as on jobs. Or we need to increase timeout :(

Reply:
This issue happening after deactivation. So I think it's not required first step.

Reply:
Technically, yes, we could simulate that with an update, but I think it is preferable to reflect the user's issue as much as possible in integration tests.

Reply:
Done